src/main/java/com/gfl/client/service/proxy/queue/DefaultProxySourceQueueHandler.java

```diff
@@ -5,25 +5,30 @@
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
-import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Queue;
 import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
 
 @Service
 public class DefaultProxySourceQueueHandler implements ProxySourceQueueHandler {
 
     private static final String COMMON_QUEUE = "common";
     private final Map<String, BlockingQueue<ProxyConfigHolder>> queues;
+    private final Map<String, Lock> queueLocks;
     private final AsyncProxyQueueTaskProcessor asyncProxyQueueTaskExecutor;
     // todo: add ProxyValidationService dependency
 
     public DefaultProxySourceQueueHandler(@Lazy AsyncProxyQueueTaskProcessor asyncProxyQueueTaskExecutor) {
         this.asyncProxyQueueTaskExecutor = asyncProxyQueueTaskExecutor;
-        this.queues = new HashMap<>();
+        this.queues = new ConcurrentHashMap<>();
+        this.queueLocks = new ConcurrentHashMap<>();
         this.queues.put(COMMON_QUEUE, new LinkedBlockingDeque<>());
+        this.queueLocks.put(COMMON_QUEUE, new ReentrantLock());
     }
 
     @Override
```

Comment:
Avoid hard-coding cron expression, consider putting it in a configuration file or property.